Multisig/UI/Settings/OwnerKeyManagement/KeystoneOwnerKey/KeystoneSignFlow.swift

```diff
@@ -101,21 +101,35 @@ extension SECP256K1.UnmarshaledSignature {
         let s = data[32..<64]
         let v: UInt8
 
-        // only if v was overflown and is legacy transaction
-        if isLegacyTx && data.count > 65 {
+        let chainIdInt = UInt64(chainId) ?? 0
+        let vBytes: Bytes
 
+        // if v was overflown (e.g. chain_id > 109 according to EIP-155)
+        if data.count > 65 {
             // max 8 bytes to fit into UInt64
-            let vBytes: Bytes = [UInt8](data.suffix(from: 64).prefix(8))
+            let vBytes = [UInt8](data.suffix(from: 64).prefix(8))
             let vInt =  UInt64(vBytes)
-            let chainIdInt = UInt64(chainId) ?? 0
-
-            // recover V by deducting (chainId * 2 + 35) according to EIP-155.
+            // recover V by deducting (chainId * 2 + 35) according to EIP-155
             let vRecovered = vInt - (chainIdInt * 2 + 35)
             v = try! UInt8(vRecovered % 256)
         } else {
-            v = data[64]
+            vBytes = [UInt8]([data[64]])
+            let vInt = UInt8(vBytes)
+            if isLegacyTx {
+                // Legacy ethereum (pre-eip-155) adds 27 to v
+                v = vInt - 27
+            } else {
+                // v still can be chainId * 2 + 35 for non-legacy transactions
+                if vInt >= 37 {
```

Comment:
this is incorrect, because chainId can be = 0 and so the eip-155 v value  will be >= 35

Reply:
this is highly unlikely though

Reply:
that is incorrect